2008-07-28  Robert Millan  <rmh@aybabtu.com>

	* disk/raid.c (grub_raid_scan_device): Do not abort when two disks
	with the same number are found, just use issue a warning with
	grub_dprintf(), as this error has been reported to be non-fatal.

Index: disk/raid.c
===================================================================
--- disk/raid.c	(revision 1741)
+++ disk/raid.c	(revision 1742)
@@ -442,16 +442,10 @@
 	}
   
       if (array->device[sb.this_disk.number] != NULL)
-	{
-	  /* We found multiple devices with the same number. Again,
-	     this shouldn't happen.*/
-
-	  grub_error (GRUB_ERR_BAD_NUMBER,
-		      "Found two disks with the number %d?!?",
+	/* We found multiple devices with the same number. Again,
+	   this shouldn't happen.*/
+	grub_dprintf ("raid", "Found two disks with the number %d?!?",
 		      sb.this_disk.number);
-
-	  return 0;
-	}
     }
 
   /* Add an array to the list if we didn't find any.  */
